pkg/apiserver/filters/auditing.go

```diff
@@ -0,0 +1,44 @@
+package filters
+
+import (
+	"k8s.io/klog"
+	"kubesphere.io/kubesphere/pkg/apiserver/auditing"
+	"kubesphere.io/kubesphere/pkg/apiserver/request"
+	"net/http"
+)
+
+func WithAuditing(handler http.Handler, a auditing.Auditing) http.Handler {
+
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+
+		if !a.Enable() {
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		info, ok := request.RequestInfoFrom(req.Context())
+		if !ok {
+			klog.Error("Unable to retrieve request info from request")
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		if info.IsKubernetesRequest {
+			handler.ServeHTTP(w, req)
+			return
+		}
+
+		e := a.LogRequestObject(info, req)
+		resp := auditing.NewResponseCapture(w)
+		go handler.ServeHTTP(resp, req)
```

Comment:
Why put handler.ServeHTTP in a goroutine, why we need to wait from resp.StopCh ?
You can refer to https://github.com/kubernetes/apiserver/blob/master/pkg/endpoints/filters/audit.go#L77 to make sure `audit LogResponseObject` is always called after finish serving requests?